### PR TITLE
ACAS-330: Improved "Open In" button UX for SEL summary

### PR DESCRIPTION
**Pull request by dalejerikson**
**Steps to Reproduce**

Load an Experiment through the Experiment Loader, and look at the summary from a successful upload.

**Expected Results**

There is a button to Generate Link which will yield a copyable LiveReport link.

The Open in LiveDesign button UX is improved (See [ACAS-277](https://jira.schrodinger.com/browse/ACAS-277))

 **Extra Information**

This was an oversight in the spec for [ACAS-272](https://jira.schrodinger.com/browse/ACAS-272) and [ACAS-277](https://jira.schrodinger.com/browse/ACAS-277). Those improvements are live on Experiment Browser but are not present in the SEL summary which is generated in R.

Brian Bolt and Brian Frost discussed a potential technical route to take here, since this fix is non-obvious.
In Brian Bolt's recent Dose Response work, he established a pattern of modifying the HTML summary after it's returned from the backend, to enhance the UX. See "BasicFileValidateAndSave.coffee" and the "handleValidationReturnSuccess" function: https://github.com/mcneilco/acas/blob/5ce299ae908b7a69cbec418d564d8bec07e823fa/modules/Components/src/client/BasicFileValidateAndSave.coffee#L161

We can take the same approach with this ticket by looking for the existing "open in" buttons within the returned HTML, then replacing them with elements like those developed in [ACAS-277](https://jira.schrodinger.com/browse/ACAS-277).
There's also a refactor we should do along the way, which is to put the HTML elements and the javascript code that [Dale Erikson](https://jira.schrodinger.com/secure/ViewProfile.jspa?name=erikson) implemented in [ACAS-277](https://jira.schrodinger.com/browse/ACAS-277) and put them into their own controller, so that we can reuse that whole controller within both Experiment Browser and this SEL summary.

## How Has This Been Tested?
Interacting w/ UI Across Various Use Cases

### CODE REVIEW

File: modules/Components/src/client/BasicFileValidateAndSave.coffee
Comment (bffrost) [inline modules/Components/src/client/BasicFileValidateAndSave.coffee:210]: @brianbolt when @dalejerikson and I were working on this, we structured the `OpenExperimentInQueryToolController` to take an experimentStatus as input because the behavior in Experiment Browser depended on the status, I think just hiding the Open In Button if the status wasn't approved and the viewer is LiveDesign.

SEL doesn't pass back an experiment status, so we didn't have access to the status without doing an additional query. It seems like the formally correct thing to do here would be to check the `server.sel.experimentStatus` config which sets the default status for things loaded through SEL, but with it being a "server" config, we can't access it from the client, right?

Just curious if you had thoughts here, or whether you think that leaving it hardcoded as "approved" is fine.

File: modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee
Comment (bffrost) [inline modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee:613]: @dalejerikson  I just remembered we had talked about refactoring a few other places in ExperimentBrowser to use this utility function. Can you add that to this PR? It feels like the right time to clean that up.

Comment (dalejerikson) [reply modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee:613]: Yep, will do!

Comment (dalejerikson) [reply modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee:564]: Completed

**Review by bffrost — APPROVED** [review]
LGTM, but I'm biased :)
Please wait to merge until you get an approval from @brianbolt

**Review by brianbolt — CHANGES_REQUESTED** [review]
Overall LGTM, just a few minor things to clean up I think.

File: modules/Components/src/client/BasicFileValidateAndSave.coffee
Comment (brianbolt) [reply modules/Components/src/client/BasicFileValidateAndSave.coffee:210]: Ignore this comment, Frost addressed it above.

File: modules/Components/src/client/OpenExperimentInQueryTool.coffee
Comment (dalejerikson) [reply modules/Components/src/client/OpenExperimentInQueryTool.coffee:80]: Yep!

File: modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee
Comment (dalejerikson) [reply modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee:363]: Yep!

File: modules/Components/src/client/OpenExperimentInQueryTool.coffee
Comment (dalejerikson) [reply modules/Components/src/client/OpenExperimentInQueryTool.coffee:47]: Will Remove. Thanks!

**Comment by dalejerikson** [comment]
All changes requested have been implemented.

**Review by bffrost — APPROVED** [review]
LGTM

**Review by brianbolt — APPROVED** [review]
LGTM

**Comment by brianbolt** [comment]
@dalejerikson Just noting that we should avoid putting links to jira in here as this is an open source project and the public can't click on those links and read the details.

**Pull request merged**
